marklogic-client-api/src/main/java/com/marklogic/client/datamovement/JacksonCSVSplitter.java

```diff
@@ -115,4 +130,29 @@ private JacksonHandle wrapJacksonHandle(JsonNode content) {
         incrementCount();
         return new JacksonHandle(content);
     }
+    
+    private void setHeaders(JsonNode headerNode) {
+        csvHeaderNode = headerNode;
+        Iterator<String> headerValue = headerNode.fieldNames();
+        this.headers = new ObjectMapper().createArrayNode();
+        while (headerValue.hasNext()) {
+            headers.add(headerValue.next());
+        }
+    }
+    
+    private PeekingIterator<JsonNode> configureSplitObj(Iterator<JsonNode> nodeItr){
+        if (!nodeItr.hasNext()) {
```

Comment:
To defend against an NPE, should the test be on nodeItr == null || !nodeItr.hasNext()?